baremetal_vm/README.md

```diff
@@ -28,74 +28,102 @@ This script creates a VM to emulate a bare-metal host, intended for developer te
+A k8s cluster must be running to host the Metal<sup>3</sup> services, this can be deployed via the [slemicro scripts](../slemicro/README.md)
 
-After the cluster is running we require some DNS names to reference the VM IP (assuming a single-node k3s cluster), so that the
-BareMetalHost VMs created in the following steps can resolve the Ironic endpoints.
+To operate with MetalLB the deployed cluster should disable the service LB e.g for k3s specify the following in your `slemicro` install `.env`
 
-This can be achieved by updating the libvirt network for the VMs (either `default` or that referred to via `VM_NETWORK`):
+```
+INSTALL_CLUSTER_EXEC="server --cluster-init --write-kubeconfig-mode=644 --disable=servicelb"
+````
+
+Additionally we require a reserved IP for MetalLB to assign for access to the provisioning services, this can be enabled by adjusting
+the DHCP range for the libvirt network, e.g using `virsh net-edit` adjust the `start` similar to below, which allows us to reserve ips under `.20`
+for static assignment:
 
 ```
-> ./get_ip.sh
-192.168.122.116
-
-> cat > /tmp/dns.xml << EOF
-<host ip='192.168.122.116'>
-  <hostname> boot.ironic.suse.baremetal </hostname>
-  <hostname> api.ironic.suse.baremetal </hostname>
-  <hostname> inspector.ironic.suse.baremetal </hostname>
-  <hostname> media.suse.baremetal </hostname>
-</host>
-EOF
-> virsh net-update default add --section dns-host --xml dns.xml --live
+  <ip address='192.168.122.1' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.122.20' end='192.168.122.254'/>
+    </dhcp>
+  </ip>
 ```
 
+Note that after editing the network you must `virsh net-destroy` then `virsh net-start` for the changes to take effect (which will disconnect
+any running VMs so they will also require a destroy/start).
+
 #### Metal<sup>3</sup> chart installation
 
 The metal3 chart can be installed with a minimal configuration via the following steps:
 
 Note this assumes that kubectl is configured e.g via the `KUBECONFIG` variable to connect to the management cluster created in the previous step.
 
-First enable the charts repo and create a values file which disables the provisioning network:
+First deploy cert-manager and MetalLB - in this example we're using `192.168.122.10` as the reserved IP for access to Ironic services:
+
 
 ```
+helm repo add jetstack https://charts.jetstack.io
 helm repo add suse-edge https://suse-edge.github.io/charts
 helm repo update
 
-cat > disable_provnet.yaml <<EOF
-global:
-  enable_dnsmasq: false
-  enable_pxe_boot: false
-  provisioningInterface: ""
-  provisioningIP: ""
-  enable_metal3_media_server: false
+helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --version v1.11.1 --set installCRDs=true
+helm install metallb suse-edge/metallb --namespace metallb-system --create-namespace
+
+export STATIC_IRONIC_IP=192.168.123.10
```

Comment:
For the future I guess we can run helm install + kustomize to deploy everything in one shot (https://helm.sh/docs/topics/advanced/#post-rendering)

Reply:
That's a good point - we probably need to discuss our preferred approach for deploying multiple charts more generally but in this case since it's primarily for development it seems reasonable to document the individual steps.

Reply:
Yep, this was just a suggestion to keep in mind for the future :)